### PR TITLE
Added support for unknown_topic_cache_ttl option in brod_client

### LINKED ISSUE #570: Why constant UNKNOWN_TOPIC_CACHE_EXPIRE_SECONDS (module brod_client) hard set 120 seconds?
I observe a strange behavior when there is no topic in Kafka. Topic status is cached. So when it’s created there is a delay up to 120 seconds. This is because of the constant -define(UNKNOWN_TOPIC_CACHE_EXPIRE_SECONDS, 120) in brod_client module. 

Is there any reason why this delay is so high? And could we make it configurable?

### PR DESCRIPTION
**Pull request by mtvch**
Resolves #570

### CODE REVIEW

File: src/brod_client.erl [hunk 1]
```diff
@@ -819,29 +821,31 @@ is_cooled_down(Ts, #state{config = Config}) ->
 
 %% call this function after fetched metadata for all topics
 %% to cache the not-found status of a given topic
-maybe_cache_unknown_topic_partition(Ets, Topic, TopicMetadataArray) ->
+maybe_cache_unknown_topic_partition(Ets, Topic, TopicMetadataArray, UnknownTopicCacheTtl) ->
   case find_partition_count_in_topic_metadata_array(TopicMetadataArray, Topic) of
     {error, unknown_topic_or_partition} = Err ->
-      _ = ets:insert(Ets, {?TOPIC_METADATA_KEY(Topic), Err, os:timestamp()}),
+      ExpireTs = erlang:monotonic_time(millisecond) + UnknownTopicCacheTtl,
```
Comment (zmstone) [inline src/brod_client.erl]: add a helper function function `expire_ts(Ttl) -> erlang:monotonic_time(millisecond) + Ttl`.

File: test/brod_client_SUITE.erl [hunk 2]
```diff
@@ -125,6 +126,19 @@ t_get_partitions_count_safe(Config) when is_list(Config) ->
   ?assertMatch({error, unknown_topic_or_partition}, Res2),
   ok = brod:stop_client(Client).
 
+
+t_get_partitions_count_configure_cache_ttl(Config) when is_list(Config) ->
+  Client = ?FUNCTION_NAME,
+  ClientConfig = [{unknown_topic_cache_ttl, 0}],
```
Comment (zmstone) [inline test/brod_client_SUITE.erl]: maybe test with 1 or 2, and inspect brod_client:lookup_partitions_count_cache changes from first call `{error, _}` to `false` after a 1/2s sleep.

File: src/brod_client.erl [hunk 3]
```diff
@@ -819,29 +821,31 @@ is_cooled_down(Ts, #state{config = Config}) ->
...
-update_partitions_count_cache(_Ets, []) -> ok;
-update_partitions_count_cache(Ets, [TopicMetadata | Rest]) ->
+-spec update_partitions_count_cache(ets:tab(), [kpro:struct()], non_neg_integer()) -> ok.
+update_partitions_count_cache(_Ets, [], _UnknownTopicCacheTtl) -> ok;
+update_partitions_count_cache(Ets, [TopicMetadata | Rest], UnknownTopicCacheTtl) ->
   Topic = kf(name, TopicMetadata),
   case get_partitions_count_in_metadata(TopicMetadata) of
     {ok, Cnt} ->
-      ets:insert(Ets, {?TOPIC_METADATA_KEY(Topic), Cnt, os:timestamp()});
+      ets:insert(Ets, {?TOPIC_METADATA_KEY(Topic), Cnt, nil});
```
Comment (zmstone) [inline src/brod_client.erl]:
```suggestion
      ets:insert(Ets, {?TOPIC_METADATA_KEY(Topic), Cnt, {added, now_ts()});
```

File: src/brod_client.erl [hunk 4]
```diff
@@ -819,29 +821,31 @@ is_cooled_down(Ts, #state{config = Config}) ->
...
     {error, ?unknown_topic_or_partition} = Err ->
-      ets:insert(Ets, {?TOPIC_METADATA_KEY(Topic), Err, os:timestamp()});
+      ExpireTs = erlang:monotonic_time(millisecond) + UnknownTopicCacheTtl,
+      ets:insert(Ets, {?TOPIC_METADATA_KEY(Topic), Err, ExpireTs});
```
Comment (zmstone) [inline src/brod_client.erl]:
```suggestion
      ets:insert(Ets, {?TOPIC_METADATA_KEY(Topic), Err, {expire, ExpireTs}});
```

**Review by zmstone — COMMENTED** [review]
Thank you for the PR.
I added a few comments inline.
Please also help to update the changlog file.

File: test/brod_client_SUITE.erl (hunk 2 above)
Comment (mtvch) [reply test/brod_client_SUITE.erl]: I changed configuration to 100ms and sleep for 101 ms to not slow down tests

**Review by zmstone — APPROVED** [review]

**Comment by zmstone** [comment]
Thank you @mtvch

**Pull request merged**